proc/merge.go

```diff
@@ -68,85 +63,17 @@ func (m *Merge) Parents() []Proc {
 	return pp
 }
 
-func (m *Merge) reload(k int) {
-	parent := m.parents[k]
-	result := <-parent.ch
-	err := result.Err
-	if err != nil && err != io.EOF {
-		// If any parent has an error, we set this error,
-		// which will cause the MergeProc to stop and return
-		// the error, and run the done protocol on all the parents.
-		m.err = err
-	}
-	buf := result.Batch
-	m.bufs[k] = buf
-	if buf == nil {
-		close(parent.proceed)
-		m.parents[k] = nil
-	} else {
-		parent.proceed <- struct{}{}
-	}
-}
-
-// fill() initializes upstream data from each parent in our
-// per-parent merge buffers.
-func (m *Merge) fill() {
-	for i := range m.parents {
-		m.reload(i)
-	}
-}
-
 // Pull implements the merge logic for returning data from the upstreams.
 func (m *Merge) Pull() (zbuf.Batch, error) {
 	m.once.Do(func() {
 		for _, m := range m.parents {
 			go m.run()
 		}
-		m.bufs = make([]zbuf.Batch, len(m.parents))
-		m.fill()
 	})
-	if m.err != nil {
-		m.Done()
-		return nil, m.err
-	}
-	oldest := nano.MaxTs
-	pick := -1
 
-	// For now our "merge" just pushes out the batch with the oldest
-	// timestamp at each round... this means that we may not
-	// sending out monotonically ordered tiemstamps. Proper
-	// time-ordered merging will come after Pull(span) is
-	// implemented.
-	for i, buf := range m.bufs {
-		if buf == nil {
-			continue
-		}
-		if buf.Span().Ts < oldest {
-			oldest = buf.Span().Ts
-			pick = i
-		}
-	}
-	if pick == -1 {
-		m.Done()
+	res, ok := <-m.ch
+	if !ok {
 		return nil, nil
 	}
-
-	res := m.bufs[pick]
-	m.reload(pick)
-	if m.err != nil {
-		m.Done()
-		return nil, m.err
-	}
-	return res, nil
-}
-
-func (m *Merge) Done() {
```

Comment:
From yesterday's slack discussion, I think I learned that Merge still needs a `Done()` call that will signal the runnerProc's to call Done() on their parent.